ui/pb/progressbar.go

```diff
@@ -97,23 +121,25 @@ func (pb *ProgressBar) Modify(options ...ProgressBarOption) {
 	}
 }
 
-// String locks the progressbar struct for reading and calls all of its methods
+// Render locks the progressbar struct for reading and calls all of its methods
 // to assemble the progress bar and return it as a string.
-//TODO: something prettier? paddings, right-alignment of the left column, line trimming by terminal size
-func (pb *ProgressBar) String() string {
+// - leftPad sets the padding between the left text and the opening
+//   square bracket.
+// - leftMax sets the maximum character length of the left text.
+//   Characters exceeding this length will be replaced with a single ellipsis.
+//   Passing <=0 disables trimming.
+func (pb *ProgressBar) Render(leftPad, leftMax int) string {
```

Comment:
hmm do you actually need both `leftPad` and `leftMax`? Can't you just determine the size of the "box" we have for the labels on the left of the progressbar? Anything shorter will be padded, anything longer will be truncated?

Reply:
Hhmm true, I did think the "left_truncated_padding" test was unnecessary, as there likely wouldn't be a reason to truncate the name, but use a different padding.

Sure, I'll make this just one value.

Reply:
Should be fixed in eece282b.